### ISSUE TITLE
@SendToKafka to send out messages to kafka topics.

### ISSUE THREAD

**Issue by ksambhav** [spring-projects/spring-kafka#239]
Will it be good to have annotation support for sending out messages to kafka topics - counter part of `@KafkaListener` annotation ? ( Or if there is one already ?! )

Something of this sort:-

```
public @interface SendToKafka {
   	String kafkaTemplate default "kafkaTemplate";  // kafka template bean to be used for sending.
	String[] topics() default {};           // topic(s) to send message to.
	TopicPartition[] topicPartitions() default {};
}

```

**Comment by artembilan** [comment]
Well, although it looks organic to be aligned with similar feature in the Spring JMS and Spring AMQP, it is pretty complicated here that there, because there is no such a `session` (or `channel`) abstraction to build `Producer` from the consumer perspective.

We may really request a `kafkaTemplate()` bean reference, but look, we have to support all the options via annotation to meet `ProducerRecord` requirements:
```
public final class ProducerRecord<K, V> {

    private final String topic;
    private final Integer partition;
    private final K key;
    private final V value;
    private final Long timestamp;

```

So, IMO it is much easier to do send manually via injected `KafkaTemplate` then try to evaluate everything via sophisticated SpEL from almost useless annotation.

In other words I don't see reason to introduce this phantom gain, since almost in all cases we would have to provide all those useful options for record.

**Comment by garyrussell** [comment]
You could however, use a Spring Integration `@MessagingGateway` with the gateway wired to a Kafka outbound channel adapter.

**Comment by artembilan** [comment]
In the upcoming version we will be able to get deal with high-level data via `KStream` API:
```
@Bean
public KStream<Integer, String> kStream(KStreamBuilder kStreamBuilder) {
	KStream<Integer, String> stream = kStreamBuilder.stream(STREAMING_TOPIC1);
		stream
			.mapValues(String::toUpperCase)
			.to(STREAMING_TOPIC2);
			return stream;
}
```

**Comment by artembilan** [comment]
I think #286 fixes this one as well.
So, now you can do like this:
```
@KafkaListener(topics = "request")
@SendTo("reply")
public Collection<String> replyingBatchListenerWithErrorHandler(List<String> in) {
```
